src/xpcc/architecture/platform/driver/i2c/stm32_i2c2_v1_1/driver.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version='1.0' encoding='UTF-8' ?>
+<!DOCTYPE rca SYSTEM "../../xml/driver.dtd">
+<rca version="1.0">
+	<driver type="i2c" name="stm32_i2c2_v1_1">
+		<!-- Template: file that needs to be turned into a source file -->
+		<template device-family="f0|f7|h7|l0|l4" instances="1,2,3" out="i2c_master_{{id}}.hpp">i2c2_v1_1_master.hpp.in</template>
```

Comment:
What is `instances="1,2,3" ` in the `driver.xml` file used for?

Reply:
The attribute specifies for which instances of a peripheral a particular source file is generated. For example, in the STM32 timer driver this feature is used to select the matching implementation for different types of timers. In most cases all possible ids should be listed. The current value "1,2,3" should probably be "1,2,3,4" to support STM32s with 4 I2Cs.

Reply:
So this information is redundant for all drivers except for the stm32 timer?!
The device xml files also specify the *instances* attribute: https://github.com/roboterclubaachen/xpcc/blob/c442b5ce1fc4630a37f2b3bf7e126ead8091f816/src/xpcc/architecture/platform/devices/stm32/stm32f405_407_415_417-i_o_r_v_z-e_g.xml#L138

Reply:
The device file specifies what instances actually exist, the driver file specifies a filter for these instances. Compare this to [the timer instance filter in modm](https://github.com/modm-io/modm/blob/develop/src/modm/platform/timer/stm32/module.lb#L60-L64).